core/trino-spi/src/main/java/io/trino/spi/connector/Connector.java

```diff
@@ -28,23 +28,14 @@
 public interface Connector
 {
     /**
-     * @deprecated use {@link #getHandleClasses()}
+     * @deprecated handle classes do not need to be predeclared anymore
```

Comment:
You can delete this method since it's not called. It will force removal when compiling if using `@Override` but will be binary compatible with any existing plugins.